tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet.md

```diff
@@ -0,0 +1,290 @@
+
+    ```
+    custom-pallet 
+    ├── Cargo.toml
+    └── src
+        └── lib.rs
+    ```
+
+    If the files are in place, your project setup is complete, and you're ready to start building your custom pallet.
+
+## Add Dependencies
+
+To build and integrate your custom pallet into a Polkadot SDK-based runtime, you must add specific dependencies to the `Cargo.toml` file of your pallet's project. These dependencies provide essential modules and features required for pallet development. Since your custom pallet is part of a workspace that includes other components, such as the runtime, the configuration must align with the workspace structure. Follow the steps below to set up your `Cargo.toml` file properly:
+
+1. Open your `Cargo.toml` file
+
+2. Add the required dependencies in the `[dependencies]` section:
+
+    ```toml
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/Cargo.toml:10:18'
+    ```
+
+3. Enable `std` features:
+
+    ```toml
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/Cargo.toml:20:27'
+    ```
+
+The final `Cargo.toml` should resemble the following:
+
+??? note "Complete `Cargo.toml` File"
+
+    ```toml
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/Cargo.toml'
+    ```
+
+## Implement the Pallet Logic
+
+In this section, you will construct the core structure of your custom pallet, starting with setting up its basic scaffold. This scaffold acts as the foundation, enabling you to later add functionality such as storage items, events, errors, and dispatchable calls.
+
+### Add Scaffold Pallet Structure
+
+You now have the bare minimum of package dependencies that your pallet requires specified in the `Cargo.toml` file. The next step is to prepare the scaffolding for your new pallet.
+
+1. Open `src/lib.rs` in a text editor and delete all the content
+   
+2. Prepare the scaffolding for the pallet by adding the following:
+
+    ```rust
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/scaffold.rs'
+    ```
+
+3. Verify that it compiles by running the following command:
+
+    ```bash
+    cargo build --package custom-pallet
+    ```
+
+### Pallet Configuration
+
+Implementing the `#[pallet::config]` macro is mandatory and sets the module's dependency on other modules and the types and values specified by the runtime-specific settings.
+
+In this step, you will configure two essential components that are critical for the pallet's functionality:
+
+- **RuntimeEvent** - since this pallet emits events, the [`RuntimeEvent`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html#associatedtype.RuntimeEvent){target=\_blank} type is required to handle them. This ensures that events generated by the pallet can be correctly processed and interpreted by the runtime
```

Comment:
```suggestion
- **`RuntimeEvent`** - since this pallet emits events, the [`RuntimeEvent`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html#associatedtype.RuntimeEvent){target=\_blank} type is required to handle them. This ensures that events generated by the pallet can be correctly processed and interpreted by the runtime
```